src/internal/sdata/sdata.go

```diff
@@ -113,61 +114,64 @@ func NewTupleFromTableInfo(info *pachsql.TableInfo) (Tuple, error) {
 	tuple := make(Tuple, len(info.Columns))
 	for i, ci := range info.Columns {
 		var err error
-		tuple[i], err = makeTupleElement(ci.DataType, ci.IsNullable)
+		tuple[i], err = makeTupleElement(ci.DataType, ci.IsNullable, ci.Precision, ci.Scale)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return tuple, nil
 }
 
-func makeTupleElement(dbType string, nullable bool) (interface{}, error) {
+func makeTupleElement(dbType string, nullable bool, precision int64, scale int64) (interface{}, error) {
 	switch dbType {
 	case "BOOL", "BOOLEAN":
 		if nullable {
 			return new(sql.NullBool), nil
-		} else {
-			return new(bool), nil
 		}
+		return new(bool), nil
 	case "SMALLINT", "INT2":
 		if nullable {
 			return new(sql.NullInt16), nil
-		} else {
-			return new(int16), nil
 		}
+		return new(int16), nil
 	case "INTEGER", "INT", "INT4":
 		if nullable {
 			return new(sql.NullInt32), nil
-		} else {
-			return new(int32), nil
 		}
-	// TODO "NUMBER" type from Snowflake can vary in precision, but default to int64 for now.
-	// https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#number
-	case "BIGINT", "INT8", "NUMBER":
+		return new(int32), nil
+	case "BIGINT", "INT8":
 		if nullable {
 			return new(sql.NullInt64), nil
-		} else {
+		}
+		return new(int64), nil
+	// FIXME: for now, use either int64 or float64 for arbitrary precision numbers,
+	// but this is not enough for numbers outside their range.
+	case "NUMERIC", "NUMBER", "DECIMAL", "FIXED":
+		if nullable {
+			if scale == 0 && precision > 0 {
+				return new(sql.NullInt64), nil
+			}
+			return new(sql.NullFloat64), nil
+		}
+		if scale == 0 && precision > 0 {
```

Comment:
I'm not totally convinced that NUMERIC without a decimal point is representable as an integer.  Consider 2^65:

```go
func main() {
        x := math.Pow(2, 65)
        fmt.Printf("%v %v\n", x, int64(x))
}
```
```
$ go run main.go
3.6893488147419103e+19 -9223372036854775808
```

I think both cases have to be `big.Rat` objects.  (I would also not try to round according to precision, let the database handle that.)